src/base/static/components/organisms/map-measurement-overlay.tsx

```diff
@@ -248,30 +248,41 @@ const MapMeasurementOverlay: React.FunctionComponent<Props> = props => {
     }
   };
 
-  React.useEffect(() => {
-    // Unless I'm missing the right way to use overlays, it seems necessary
-    // to attach a click listener here to capture clicks on the overlay.
-    // See: https://github.com/uber/react-map-gl/issues/470
-    if (overlayRef && overlayRef.current && overlayRef.current._canvas) {
-      overlayRef.current._canvas.addEventListener("click", handleOverlayClick);
-      // This direct manipulation of the `pointer-events` style property feels
-      // like a hack, but the goal is to enable clicks on the underlying map
-      // when no measurement tool is selected.
-      overlayRef.current._canvas.style.pointerEvents = "none";
-    } else {
-      // eslint-disable-next-line no-console
-      console.warn(
-        "Measurement overlay: failed to attach click event listener",
-      );
-    }
+  React.useEffect(
+    () => {
+      // Unless I'm missing the right way to use overlays, it seems necessary
+      // to attach a click listener here to capture clicks on the overlay.
+      // See: https://github.com/uber/react-map-gl/issues/470
+      if (
+        props.isMeasurementToolVisible &&
+        overlayRef &&
+        overlayRef.current &&
+        overlayRef.current._canvas
+      ) {
+        overlayRef.current._canvas.addEventListener(
```

Comment:
One thing I'm uncertain about here is whether there is a better way to attach this listener. Because we now can return `null` from this component (depending on whether it's visible or not), we have to be careful to attach a click listener after a non-null render, so the ref to the canvas element is available.

This code will re-run each time the overlay tools are hidden and re-rendered, since `props.isMeasurementToolVisible` is a "dependency" of this `useEffect` callabck. I was worried this would cause multiple click listeners to build up, but this does not actually happen. I *think* it's because this component returns `null` when `props.isMeasurementToolVisible` is `false`, removing existing listeners. But I haven't been able to find a specific explanation of what happens in this situation.

Reply:
I will go ahead and merge this for now though...